[DTensor] ExplicitRedistributionContext warning mode

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #169452

Allow issuing warnings instead of raising exceptions.

- easier to capture all issues in one pass

wconstab committed:
[DTensor] ExplicitRedistributionContext warning mode

Allow issuing warnings instead of raising exceptions.

- easier to capture all issues in one pass

[ghstack-poisoned]

wconstab committed:
Update on "[DTensor] ExplicitRedistributionContext warning mode"


Allow issuing warnings instead of raising exceptions.

- easier to capture all issues in one pass

[ghstack-poisoned]

wconstab committed:
Update on "[DTensor] ExplicitRedistributionContext warning mode"


Allow issuing warnings instead of raising exceptions.

- easier to capture all issues in one pass

[ghstack-poisoned]

zpcore commented on torch/distributed/tensor/_utils.py:
```diff
@@ -29,22 +33,41 @@ class ExplicitRedistributionContext:
+    strict (bool) if True, triggers on any redistribution.  If False, only triggers on redistributions that perform
+    communication.
+
+    mode (str) "raise": raises an
```
sentence not completed

wconstab committed:
Update on "[DTensor] ExplicitRedistributionContext warning mode"


Allow issuing warnings instead of raising exceptions.

- easier to capture all issues in one pass

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge
